main.tf

```diff
@@ -35,6 +35,13 @@ resource "google_bigquery_dataset" "main" {
   project                     = var.project_id
   labels                      = var.dataset_labels
 
+  dynamic "default_encryption_configuration" {
+    for_each = (var.dataset_default_encryption_key == null || var.dataset_default_encryption_key == "null") ? [] : list(var.dataset_default_encryption_key)
```

Comment:
We dont usually check for `"null"` string. Expectation is that the user let it default to `null` or explicitly set a value. is there a reason for this check?

Reply:
The reason is to allow the user to use "null" as a valid input to clear the field, vs removing the setting from composition. This allows a flag in the module's composition to be set, for example:

dataset_default_encryption_key = var.enable_cmek == "true" ? var.kms_keyring : null

Reply:
You are right here, I am mixing concerns... going to remove this.

Reply:
@bharathkkb here is the concern on the string eval. When I pass in

dataset_default_encryption_key = var.kms_key

In examples/multiple_tables/main.tf; if the var is at default the null string is interpreted as the key and attempts to configure. So if we define and set the var.kms_key's default (null), it's a string input.

Reply:
@kpeder That's odd. I don't think that should be the case but even if it does as `tostring(null) == null`. I will try a test on my end.

Reply:
I tested it by removing `|| var.dataset_default_encryption_key == "null"` and then trying to use `examples/multiple_tables` without an input for `kms_key`. It seems to work as expected. Terraform v0.13.5
```hcl
  # module.bigquery.google_bigquery_dataset.main will be created
  + resource "google_bigquery_dataset" "main" {
      + creation_time               = (known after apply)
      + dataset_id                  = "foo"
      + default_table_expiration_ms = 3600000
      + delete_contents_on_destroy  = true
      + description                 = "some description"
      + etag                        = (known after apply)
      + friendly_name               = "foo"
      + id                          = (known after apply)
      + labels                      = {
          + "billable" = "true"
          + "env"      = "dev"
          + "owner"    = "janesmith"
        }
      + last_modified_time          = (known after apply)
      + location                    = "US"
      + project                     = "project-id"
      + self_link                   = (known after apply)

      + access {
          + role          = "OWNER"
          + special_group = "projectOwners"
        }
    }
```

Reply:
I get the same result. I think I must have passed it a string in at some point instead of a 'naked' null; the original test didn't have the default, and I think the python-generated outputs made it a string. The jsondecode() was similarly needed after the python step.